### PR TITLE
docs: add bare-metal install example yaml

### PR DESCRIPTION
**Pull request by dave93cab**
I know there is an example at the top of the page but the example mid-page was empty yaml

### CODE REVIEW

**Comment by andrewrynhard** [comment]
Thanks @dave93cab! A few things to note so we can get this in.

1. There PR must have only 1 commit (a squash is needed)
2. The commit must be in conventional commit format: https://www.conventionalcommits.org/en/v1.0.0/. So in this case something like `docs: add bare-metal install example yaml` would work.
3. The edits you made are actually generated from go code. I think you might actually want to edit https://github.com/talos-systems/talos/blob/master/pkg/config/types/v1alpha1/v1alpha1_types.go#L132-L142 instead, and then run `make docs`.

Hope that is clear, and helpful. Feel free to ping me if you have any questions.

**Comment by dave93cab** [comment]
That only update 0.4 docs

**Comment by andrewrynhard** [comment]
@dave93cab Yeah, I can make a similar PR for v0.3 (the `release-0.3`) docs. We're still working on our tooling to simplify this process, so unfortunately I will have to make a PR against that branch as well.

File: pkg/config/types/v1alpha1/v1alpha1_types.go [hunk 1]
```diff
@@ -134,12 +134,13 @@ type MachineConfig struct {
...
 	//     - |
 	//       install:
-	//         disk:
+	//         disk: /dev/sda
 	//         extraKernelArgs:
-	//         image:
-	//         bootloader:
-	//         wipe:
-	//         force:
+        //           - option=value
```
Comment (andrewrynhard) [inline pkg/config/types/v1alpha1/v1alpha1_types.go]: looks like an extra indent here that makes `gofmt` unhappy.

**Comment by andrewrynhard** [comment]
@dave93cab looks like I missed something too ... the commit needs a DCO. You can amend the commit with the `-s` flag to include the DCO. `git commit -s --amend --no-edit`.

**Comment by dave93cab** [comment]
@andrewrynhard hopefully that is it. Thanks for your patience

**Pull request merged**